src/cuda-sha256/sha256.h

```diff
@@ -0,0 +1,25 @@
+#include "tomcrypt_macros.h"
+
+
+#ifndef SHA256_H
+#define SHA256_H
+
+#define SHA256_BLOCK_SIZE 32
+
+struct sha256_state {
+    ulong64 length;
+    ulong32 state[8], curlen;
+    unsigned char buf[64];
+};
+
+typedef struct {
+    struct sha256_state sha256;
+} hash_state;
+
+void __host__ __device__ sha256_init(hash_state*);
```

Comment:
These will not be inlined, no? can you see if that has an affect on perf?

Reply:
No, I don't think it will be inlined as is, but does adding the `.h` file change that? I have't gotten to perf and benchmarking yet, I wanted to test correctness with this first.

Reply:
Yea. it will affect the code-gen for the chacha code.

Reply:
We should test the correctness before merging no?

Reply:
I've done local correctness testing, but I'd like to get this going through solana CI for a more thorough test.